core/src/main/java/org/elasticsearch/index/query/BoolQueryBuilder.java

```diff
@@ -294,23 +294,25 @@ public String getName() {
 
     @Override
     public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
-        if (!hasClauses()) {
-            return new MatchAllDocsQuery();
-        }
-
-        BooleanQuery booleanQuery = new BooleanQuery(disableCoord);
-        addBooleanClauses(parseContext, booleanQuery, mustClauses, BooleanClause.Occur.MUST);
-        addBooleanClauses(parseContext, booleanQuery, mustNotClauses, BooleanClause.Occur.MUST_NOT);
-        addBooleanClauses(parseContext, booleanQuery, shouldClauses, BooleanClause.Occur.SHOULD);
-        addBooleanClauses(parseContext, booleanQuery, filterClauses, BooleanClause.Occur.FILTER);
-
-        booleanQuery.setBoost(boost);
-        Queries.applyMinimumShouldMatch(booleanQuery, minimumShouldMatch);
-        Query query = adjustPureNegative ? fixNegativeQueryIfNeeded(booleanQuery) : booleanQuery;
-        if (queryName != null) {
-            parseContext.addNamedQuery(queryName, query);
+        Query returnedQuery = new MatchAllDocsQuery();
+        if (hasClauses()) {
```

Comment:
if it doesn't have clasues we don't set the boost and ignore the queryName. I think it would be a bit more readable if we had two if branches, and the set boost and named query handling after the if which kicks in in both cases.